### PR TITLE
[Slack] 👻 Add Ephemeral Message Support

### PR DESCRIPTION
**Pull request by jonchurch**
First pass at this for everyone hungry to jump on the new features! 
### **TLDR; We have bot.whisper now!!**

This adds two new functions to Botkit to make use of the new [`chat.postEphemeral`](https://api.slack.com/methods/chat.postEphemeral) Web API method.

### **bot.whisper()**:
Works just like bot.reply, takes a source message and clones the channel and user (user is required for this api method) and sends it. We also set `as_user` to true to show the messages are coming from the bot user (or whoevers token you're making calls with). This can be overridden by setting `user: false` on the outgoing message body, or using the second new method:

### **bot.sendEphemeral()**:
Works like bot.send, takes a botkit message with a channel and user (again, user is required for this method by slack) and sends a message, unprompted by the user.

## Ephemeral in Conversations
~Currently there is shakey support of interactive messages~, **slack is not sending the original message in the trigger payload, which is normally sent along with interactive message payloads.** ~replyInteractive to ephemeral interactive messages doesnt update them either. I asked in the slack api channel but havent heard anything about this yet.~  

~That unfortunately makes ephemeral messages, at this time, not as feature rich as their standard counterparts, and doing complex user flows with updating their choices isnt possible currently in the ephemeral realm 👻~

Got a response from @episod! Interactive messages do work with ephem. However, slack is not passing the original message object along with the payload. To respond, you must construct a new message. I've experienced issues where my updated message is attributed to the App and not the bot. Not sure what I'm doing wrong still, passing `as_user: true` doesnt seem to work.

**To send an ephemeral message as part of a normal conversation, add `ephemeral: true` to the message body. See the updated docs for an example**

### CODE REVIEW

File: lib/Slackbot_worker.js [hunk 1]
```diff
@@ -338,6 +337,10 @@ module.exports = function(botkit, config) {
     };
 
     bot.send = function(message, cb) {
+		if (message.ephemeral) {
```
Comment (jonchurch) [inline lib/Slackbot_worker.js]: I chose to retain a single send function, and just check if the message should be ephemeral or not

Using send like this also preserves access to these messages from send middlewares!

File: lib/Slackbot_worker.js [hunk 2]
```diff
@@ -413,6 +416,39 @@ module.exports = function(botkit, config) {
...
+            channel: message.channel,
+            text: message.text || null,
+            user: message.user,
+            as_user: message.as_user || false,
+            parse: message.parse || null,
+            link_names: message.link_names || null,
+            attachments: message.attachments ?
+                JSON.stringify(message.attachments) : null,
+        };
+		console.log({message})
```
Comment (k001) [inline lib/Slackbot_worker.js]: Why not use `botkit.debug` ?

File: lib/Slackbot_worker.js [hunk 3]
```diff
@@ -413,6 +416,39 @@ module.exports = function(botkit, config) {
...
+		console.log({slack_message})
```
Comment (k001) [inline lib/Slackbot_worker.js]: Same here, is better use `botkig.debug`

File: lib/Slackbot_worker.js (hunk 2 above)
Comment (jonchurch) [reply lib/Slackbot_worker.js]: No good reason, ty for catching it
